core/src/utilities/random.cc

```diff
@@ -746,4 +950,136 @@ namespace jiminy
             }
         };
     }
-}
\ No newline at end of file
+
+    template<typename PerlinProcessType>
+    void computePerlinHeightmap(PerlinProcessType perlinProcess,
```

Comment:
Just make this signature more general to support specifying pairs `(axis, PerlinProcess)`, this way:
```
    template<typename PerlinProcessType, int N>
    void computePerlinHeightmap(std::array<std::pair<Eigen::Vector3, PerlinProcessType>, N> perlinProcesses,
                                ...
```

Reply:
Then all you have to do is to project the current pose only each axes and aggregate the result. You can do this automatically using `sumHeightmaps`, which has been designed for this exact purpose.

Reply:
Like shown in the photo above, I don't think that's what we want.